File: 125-ValidPalindrome/125-ValidPalindrome.c
// Last updated: 1/11/2026, 12:37:15 PM
bool isPalindrome(char* s) {
    #include <stdio.h>
    #include <string.h>
    #include <ctype.h>

    int left = 0;
    int right = strlen(s) - 1;

    while (left < right) {
        while (left < right && !isalnum(s[left])) {
            left++;
        }
        while (left < right && !isalnum(s[right])) {
            right--;
        }
        if (tolower(s[left]) != tolower(s[right])) {
            return false;
        }
        left++;
        right--;
    }

    return true;
}

File: 125-ValidPalindrome/125-ValidPalindrome.py
# Last updated: 1/11/2026, 12:37:19 PM
class Solution:
    def isPalindrome(self, s: str) -> bool:
        clean = ""
        for i in s:
            if i.isalnum():
                clean += i
        
        clean = clean.lower()
        
        if clean == clean[::-1]:
            return True
        
        else:
            return False